pwnlib/gdb.py

```diff
@@ -750,7 +750,10 @@ def binary():
         >>> gdb.binary() # doctest: +SKIP
         '/usr/bin/gdb'
     """
-    gdb = misc.which('pwntools-gdb') or misc.which('gdb')
+    if context.gdb_binary:
+        gdb = misc.which(context.gdb_binary)
+    else:
+        gdb = misc.which('pwntools-gdb') or misc.which('gdb')
 
     if not context.native:
         multiarch = misc.which('gdb-multiarch')
```

Comment:
I think we should always use the gdb binary set in `context.gdb_path` and assume the user knows what they want.

Reply:
Good catch, fixed by returning early if it's set. It'll also log a warning if the path is not found.